test/vm_tests/builder/conditionally_null_safe_builder_test.dart

```diff
@@ -0,0 +1,97 @@
+            'test_fixtures/test_packages/non_null_safe/.dart_tool/build/generated/over_react__test__non_null_safe/lib/test.over_react.g.dart');
+        expect(expectedGeneratedFile.existsSync(), isTrue,
+            reason:
+                'File should have been generated: ${expectedGeneratedFile.path}');
+        expect(
+            expectedGeneratedFile.readAsStringSync(),
+            contains(
+                'Using nullSafety: false. {languageVersion: 2.11, source: packageConfig, package: over_react__test__non_null_safe}'));
+      });
+
+      test('but has a null safe language version comment', () {
+        final expectedGeneratedFile = File(
+            'test_fixtures/test_packages/non_null_safe/.dart_tool/build/generated/over_react__test__non_null_safe/lib/test_language_version_comment.over_react.g.dart');
+        expect(expectedGeneratedFile.existsSync(), isTrue,
+            reason:
+                'File should have been generated: ${expectedGeneratedFile.path}');
+        expect(
+            expectedGeneratedFile.readAsStringSync(),
+            contains(
+                'Using nullSafety: true.  {languageVersion: 2.13, source: libraryVersionComment}'));
+      });
+    });
+
+    group('on package-level Dart version that is null safe', () {
+      setUpAll(() {
+        const nonNullSafePackagePath = 'test_fixtures/test_packages/null_safe';
+        Process.runSync('dart', ['pub', 'get'],
+            workingDirectory: nonNullSafePackagePath);
+        final build = Process.runSync(
+            'dart', ['pub', 'run', 'build_runner', 'build'],
+            workingDirectory: nonNullSafePackagePath);
+
+        expect(build.exitCode, 0);
+      });
+
+      test('', () {
+        final expectedGeneratedFile = File(
+            'test_fixtures/test_packages/null_safe/.dart_tool/build/generated/over_react__test__null_safe/lib/test.over_react.g.dart');
+        expect(expectedGeneratedFile.existsSync(), isTrue,
+            reason:
+                'File should have been generated: ${expectedGeneratedFile.path}');
+        expect(
+            expectedGeneratedFile.readAsStringSync(),
+            contains(
+                'Using nullSafety: true.  {languageVersion: 2.12, source: packageConfig, package: over_react__test__null_safe}'));
+      });
+
+      test('but has a non-null safe language version comment', () {
+        final expectedGeneratedFile = File(
+            'test_fixtures/test_packages/null_safe/.dart_tool/build/generated/over_react__test__null_safe/lib/test_language_version_comment.over_react.g.dart');
+        expect(expectedGeneratedFile.existsSync(), isTrue,
+            reason:
+                'File should have been generated: ${expectedGeneratedFile.path}');
+        expect(
+            expectedGeneratedFile.readAsStringSync(),
+            contains(
+                'Using nullSafety: false. {languageVersion: 2.11, source: libraryVersionComment}'));
+      });
```

Comment:
Since the over_react builder builds for the current package and all dependencies, I think it's important for us to have test explicit test coverage that the correct language version is detected for code in dependencies.

The `over_react__test__null_safe` package depends on `over_react__test__non_null_safe`, so could we validate that those files are still built correctly when building this package? Should be the same expectations as above, but within this build.